server/services/v1/database/base_runner.go

```diff
@@ -246,37 +246,37 @@ func (runner *BaseQueryRunner) getWriteIterator(ctx context.Context, tx transact
 	collection *schema.DefaultCollection, reqFilter []byte, collation *value.Collation,
 	metrics *metrics.WriteQueryMetrics,
 ) (Iterator, error) {
-	var (
-		err      error
-		iKeys    []keys.Key
-		iterator Iterator
-	)
-
 	reader := NewDatabaseReader(ctx, tx)
 
-	if iKeys, err = runner.buildKeysUsingFilter(collection, reqFilter, collation); err == nil {
-		iterator, err = reader.KeyIterator(iKeys)
-	} else {
-		if iterator, err = reader.ScanTable(collection.EncodedName); err != nil {
-			return nil, err
-		}
-		filterFactory := filter.NewFactory(collection.QueryableFields, collation)
-		var filters []filter.Filter
-		if filters, err = filterFactory.Factorize(reqFilter); err != nil {
-			return nil, err
+	if iKeys, err := runner.buildKeysUsingFilter(collection, reqFilter, collation); err == nil {
+		if iterator, err := reader.KeyIterator(iKeys); err == nil {
+			metrics.SetWriteType("non-pkey")
+			return iterator, nil
 		}
+	}
 
-		iterator, err = reader.FilteredRead(iterator, filter.NewWrappedFilter(filters))
+	pkKey, err := runner.buildPrimaryKey(collection)
+	if err != nil {
+		return nil, err
 	}
+
+	pkIterator, err := reader.ScanPrimaryKeyTable(pkKey)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(iKeys) == 0 {
-		metrics.SetWriteType("pkey")
-	} else {
-		metrics.SetWriteType("non-pkey")
+	filterFactory := filter.NewFactory(collection.QueryableFields, collation)
+	var filters []filter.Filter
+	if filters, err = filterFactory.Factorize(reqFilter); err != nil {
+		return nil, err
 	}
 
+	iterator, err := reader.FilteredRead(pkIterator, filter.NewWrappedFilter(filters))
+
+	metrics.SetWriteType("pkey")
 	return iterator, nil
 }
+
+func (runner *BaseQueryRunner) buildPrimaryKey(coll *schema.DefaultCollection) (keys.Key, error) {
+	return runner.encoder.EncodeKey(coll.EncodedName, coll.Indexes.PrimaryKey, []interface{}{})
```

Comment:
do we need to pass an empty slice at the end?

Reply:
Sadly yes. Otherwise it errors